crates/bevy_graph/src/graphs/list/multi.rs

```diff
@@ -0,0 +1,586 @@
+
+impl<N, E, const DIRECTED: bool> Graph<N, E> for MultiListGraph<N, E, DIRECTED> {
+    fn new() -> Self {
+        Self {
+            nodes: HopSlotMap::with_key(),
+            edges: HopSlotMap::with_key(),
+            adjacencies: SecondaryMap::new(),
+        }
+    }
+
+    fn with_capacity(node_capacity: usize, edge_capacity: usize) -> Self {
+        Self {
+            nodes: HopSlotMap::with_capacity_and_key(node_capacity),
+            edges: HopSlotMap::with_capacity_and_key(edge_capacity),
+            adjacencies: SecondaryMap::with_capacity(node_capacity),
+        }
+    }
+
+    #[inline]
+    fn capacity(&self) -> (usize, usize) {
+        (self.nodes.capacity(), self.edges.capacity())
+    }
+
+    #[inline]
+    fn node_capacity(&self) -> usize {
+        self.nodes.capacity()
+    }
+
+    #[inline]
+    fn edge_capacity(&self) -> usize {
+        self.edges.capacity()
+    }
+
+    #[inline]
+    fn reserve_nodes(&mut self, additional: usize) {
+        self.nodes.reserve(additional);
+    }
+
+    #[inline]
+    fn reserve_edges(&mut self, additional: usize) {
+        self.edges.reserve(additional);
+    }
+
+    #[inline]
+    fn is_directed(&self) -> bool {
+        DIRECTED
+    }
+
+    #[inline]
+    fn is_multigraph(&self) -> bool {
+        true
+    }
+    #[inline]
+    fn node_count(&self) -> usize {
+        self.nodes.len()
+    }
+
+    #[inline]
+    fn edge_count(&self) -> usize {
+        self.edges.len()
+    }
+
+    fn add_node(&mut self, node: N) -> NodeIdx {
+        let idx = self.nodes.insert(node);
+        let storage = if DIRECTED {
+            AdjacencyStorage::Directed(Vec::new(), Vec::new())
+        } else {
+            AdjacencyStorage::Undirected(Vec::new())
+        };
+        self.adjacencies.insert(idx, storage);
+        idx
+    }
+
+    fn try_add_edge(
+        &mut self,
+        src: NodeIdx,
+        dst: NodeIdx,
+        value: E,
+    ) -> Result<EdgeIdx, GraphError> {
+        if !self.contains_node(src) {
+            Err(GraphError::NodeNotFound(src))
+        } else if !self.contains_node(dst) {
+            Err(GraphError::NodeNotFound(dst))
+        } else {
+            unsafe {
+                let idx = self.edges.insert(Edge(src, dst, value));
+                self.adjacencies
+                    .get_unchecked_mut(src)
+                    .outgoing_mut()
+                    .get_value_or_default_mut(dst)
+                    .push(idx);
+                self.adjacencies
+                    .get_unchecked_mut(dst)
+                    .incoming_mut()
+                    .get_value_or_default_mut(src)
+                    .push(idx);
+                Ok(idx)
+            }
+        }
+    }
+
+    #[inline]
+    fn contains_node(&self, node: NodeIdx) -> bool {
+        self.nodes.contains_key(node)
+    }
+
+    fn contains_edge_between(&self, src: NodeIdx, dst: NodeIdx) -> bool {
+        match self.adjacencies.get(src).unwrap().outgoing().get_value(dst) {
```

Comment:
Oh wait I unwrapped here. I wonder what I should do in that situation.. is unwrapping fine? I don't really want to return a Result, do I? should it be false?